projects/plugins/social/src/js/components/social-notes-toggle/index.tsx

```diff
@@ -15,6 +15,19 @@ type SocialNotesToggleProps = {
 	disabled?: boolean;
 };
 
+const handleStateUpdating = ( updateFunction, updatingStateSetter, ...args ) => {
+	// Set the updating state to true
+	updatingStateSetter( true );
+	document.body.style.cursor = 'wait';
+	// Call the updateFunction with provided arguments
+	const promise = updateFunction( ...args );
+	// When the promise resolves (update is completed), set the updating state to false
+	promise.then( () => {
+		updatingStateSetter( false );
+		document.body.style.cursor = 'auto';
+	} );
```

Comment:
`updatingStateSetter( false )` should ideally happen in `finally ()` block just in case the API call fails.

Reply:
```suggestion
	promise.finally( () => {
		updatingStateSetter( false );
		document.body.style.cursor = 'auto';
	} );
```

Reply:
Yeah good point, thanks